usr/bin/unpack-image

```diff
@@ -51,33 +66,52 @@ shift $((OPTIND - 1))
 [[ "$EUID" -ne 0 ]] && die "must be run as root"
 [[ -d "$UPDATE_DIR" ]] || die "$UPDATE_DIR does not exist"
 
-report_progress_inc 10 "Extracting upgrade image."
-
 case "$1" in
 *.upgrade.tar.gz) ;;
 *) die "The upgrade image must be a '.upgrade.tar.gz' file" ;;
 esac
 
+UPGRADE_IMAGE_PATH="$(readlink -f "$1")"
+[[ -n "$UPGRADE_IMAGE_PATH" ]] || die "unable to determine upgrade image path"
+
 trap cleanup EXIT
 UNPACK_DIR=$(mktemp -d -p "$UPDATE_DIR" -t unpack.XXXXXXX)
-[[ -d "$UNPACK_DIR" ]] || die "failed to create unpack directory"
+[[ -d "$UNPACK_DIR" ]] || die "failed to create unpack directory '$UNPACK_DIR'"
+pushd "$UNPACK_DIR" &>/dev/null || die "'pushd $UNPACK_DIR' failed"
+
+report_progress_inc 10 "Extracting upgrade image."
 
-(
-	cd "$UNPACK_DIR"
-	tar -xzf -
-) <"$1" ||
-	die "failed to extract upgrade image '$1'"
+tar -xzf "$UPGRADE_IMAGE_PATH" ||
+	die "failed to extract upgrade image '$UPGRADE_IMAGE_PATH'"
 
 report_progress_inc 40 "Verifying format."
 
-[[ -f "$UNPACK_DIR/version.info" ]] || die "the upgrade image is corrupt"
+for file in SHA256SUMS payload.tar.gz version.info; do
+	[[ -f "$file" ]] || die "image is corrupt; missing '$file' file"
+done
+
+if ! $opt_s; then
+	openssl dgst -sha256 \
+		-verify /var/lib/delphix-appliance/key-public.pem.upgrade.5.3 \
+		-signature SHA256SUMS.sig.5.3 \
+		SHA256SUMS >/dev/null ||
+		die "image is corrupt; verification of SHA256SUMS file failed"
```

Comment:
Nit, can we use msg string that specifies failure was in signature validation? maybe "image is corrupt; verification of signature $sig_file failed"

Reply:
since all the filenames are currently hard coded, I feel like that's unnecessary.. but sure, I can add the filenames to the message.